### DISCUSSION TITLE
[PLEASE TEST!] inReach MapShare Version 2.0.0 🚀

### DISCUSSION THREAD

**Discussion by morehawes** [OpenGIS/inreach-mapshare#10] [Announcements]
I am happy to announce version 2 of the plugin! 🎉

As this release involved some major changes, **it is currently on the [staging](https://github.com/OpenGIS/inreach-mapshare/tree/staging) branch** while any bugs are ironed out.

You can download the plugin zip [here](https://github.com/OpenGIS/inreach-mapshare/archive/refs/tags/2.0.0.zip), which can be uploaded directly to WordPress for testing.

Please let me know what you think!

Cheers,

Joe

## Changes:

- **Refactor** - Major plugin overhaul, mostly focusing on simplifying the build process and improving the development experience. Grunt has been removed and [Vite](https://vitejs.dev/) is now used to build plugin assets.
- **Tracking Precision** - Select the precision level (Low/Medium/High) of coordinates used to build the Map. By default, Garmin provides values to a precision of 10cm, lowering the level of precision will effectively fuzz locations, making them more generalised. This Setting is available in Settings > inReach MapShare > Advanced. Thanks to [mmcc-xx](https://github.com/mmcc-xx) for the [suggestion](https://github.com/OpenGIS/inreach-mapshare/issues/5) and for working on the implementation.
- **Detail Panel** - A button has been added to the Map to collapse/expand the Detail Panel in order to increase/decrease the size of the Map. A Setting has also been added to specify if the panel is expanded by default (Settings > inReach MapShare > Appearance). Thanks to [mmcc-xx](https://github.com/mmcc-xx) for the [suggestion](https://github.com/OpenGIS/inreach-mapshare/issues/9).
- **Routes**
  - Added Route URL to Shortcode Generator (Settings > inReach MapShare), allowing users to display a planned route on the Map.
  - Added a Setting to specify the colour of the route (Settings > inReach MapShare > Appearance).
  - If a Route is provided, the Map will now zoom to fit the bounds of the route **and** any track points.
- Minor UI improvements and bug fixes.

## Installation

1. [Download the plugin (.zip)](https://github.com/OpenGIS/inreach-mapshare/archive/refs/tags/2.0.0.zip).
2. In your WordPress Admin, go to Plugins > Add New Plugin.
3. Click the "Upload Plugin" button.
4. Upload the `inreach-mapshare.zip` file (**MUST BE THIS EXACT NAME!!!**)
5. Activate the plugin.
6. Configure the plugin in WP Admin > Settings > inReach MapShare.

**Comment by diannevolek** [comment]
I came across your Wayfair plugin, which got me to your website! Arizona looks amazing! Thank you for your WordPress work - it's really elegant.

**Comment by yeti08** [comment]
I haven't been able to get this version to work, upgrading from 1.3. I renamed the file, but I think the zipped folder shouldn't have the -2.0.0 in it for the plugin to work. I'm hiking the Pacific Crest Trail with only a phone so I can't do much more troubleshooting, so will revert to 1.3 for now. Thank you much got your efforts on this!

**Reply by morehawes** [reply]
Happy trails! :D

**Comment by Kyleblythewhite** [comment]
Hoya mate, I hope you are well.

is it possible to show different embeds on different pages with different short codes just using the mapshare ID? 

I have a site that needs to track some boats and every captains got a garmin in recah - we are currently doing fine using gamins own embed but i prefer the look and feel and creative edibility of yours… would be great to know if this is possible as id look to swap them out.

i tried with version 1.3 but it only seemed to allow one embed on the website before… 

Has this changed? And if not is it something you would consider? :) 

thanks mate

kyle

**Reply by morehawes** [reply]
Thanks for reaching out. This should be possible :-/

> i tried with version 1.3 but it only seemed to allow one embed on the website before…

What happened when you tried this? Can you provide a link to illustrate the outcome?

I currently do not have an active inReach subscritpion, so it is hard for me to test one feed, let alone multiple ones.

If you are able to provide the MapShare identifiers then that would be really useful for testing.

Cheers,

Joe
